xplique/attributions/image_object_detector_explainer.py

```diff
@@ -0,0 +1,214 @@
+"""
+Module related to Object detector method
+"""
+from typing import Iterable, Tuple, Union, Optional
+import abc
+import tensorflow as tf
+import numpy as np
+from xplique.attributions.base import BlackBoxExplainer
+from xplique.commons.tf_operations import inference_batching
+
+
+class IIouCalculator:
+    """
+    Used to compute the IOU.
+    """
+    @abc.abstractmethod
+    def intersect(self, obj1: tf.Tensor, obj2: tf.Tensor) -> tf.Tensor:
+        """
+        Compute the intersection between two batched objects
+        An object could be box, segmentation mask...
+        :param obj1: object1
+        :param obj2: object2
+        :return: a score between [0,1] corresponding to the intersection of the 2 objects.
+        """
+        raise NotImplementedError()
+
+
+class SegmentationIouCalculator(IIouCalculator):
+    """
+    Used to compute the Mask IOU.
+    """
+    def intersect(self, mask1: tf.Tensor, mask2: tf.Tensor) -> tf.Tensor:
+        """
+        Compute the intersection between two batched segmentation mask.
+        The segmentation is a boolean mask on the whole image
+        :param mask1: mask of segmentation 1
+        :param mask2: mask of segmentation 2
+        :return: iou score
+        """
+        # pylint: disable=W0237
+        axis = np.arange(1, len(tf.shape(mask1)))
+        inter_area = tf.reduce_sum(tf.cast(tf.logical_and(mask1, mask2),
+                                    dtype=tf.float32), axis=axis)
+        union_area = tf.reduce_sum(tf.cast(tf.logical_or(mask1, mask2),
+                                    dtype=tf.float32), axis=axis)
+
+        return inter_area / tf.maximum(union_area, 1.0)
+
+
+
+class BoxIouCalculator(IIouCalculator):
+    """
+    Used to compute the Bounding Box IOU.
+    """
+    EPSILON = tf.constant(1e-4)

```

Comment:
Maybe it could be interesting to be able to change this parameter ? It might be interesting to initialize the BoxIouCalculator with this epsilon as a parameter ?

Reply:
It's just a tiny value in case of division by zero.
(Cf. Rise implementation for example)

Reply:
👍